pandas/core/window/rolling.py

```diff
@@ -243,7 +243,9 @@ def _create_data(self, obj: FrameOrSeries) -> FrameOrSeries:
         if self.on is not None and not isinstance(self.on, Index):
             if obj.ndim == 2:
                 obj = obj.reindex(columns=obj.columns.difference([self.on]), copy=False)
-
+        if self.axis == 1:
+            obj = obj.astype("float64", copy=False)
```

Comment:
Could you add a comment with the issue number and a small explanation on why we are doing this?

Reply:
Done

Reply:
cc @jbrockmendel FYI